packages/git-diff/lib/git-diff-view.js

```diff
@@ -1,64 +1,143 @@
 
-module.exports = class GitDiffView {
+/**
+ * @describe Handles per-editor event and repository subscriptions.
+ * @param editor {Atom.TextEditor} - The editor this view will manage.
+ */
+export default class GitDiffView {
   constructor(editor) {
-    this.updateDiffs = this.updateDiffs.bind(this);
-    this.editor = editor;
+    // These are the only members guaranteed to exist.
     this.subscriptions = new CompositeDisposable();
-    this.decorations = {};
-    this.markers = [];
-  }
+    this.editor = editor;
+    this.repository = null;
+    this.markers = new WeakMap();
 
-  start() {
-    const editorElement = this.editor.getElement();
+    // I know this looks janky but it works. Class methods are available
+    // before the constructor is executed. It's a micro-opt above lambdas.
+    const subscribeToRepository = this.subscribeToRepository.bind(this);
+    // WARNING: This gets handed to requestAnimationFrame, so it must be bound.
+    this.updateDiffs = this.updateDiffs.bind(this);
 
-    this.subscribeToRepository();
+    subscribeToRepository();
 
     this.subscriptions.add(
-      this.editor.onDidStopChanging(this.updateDiffs),
-      this.editor.onDidChangePath(this.updateDiffs),
-      atom.project.onDidChangePaths(() => this.subscribeToRepository()),
-      atom.commands.add(editorElement, 'git-diff:move-to-next-diff', () =>
-        this.moveToNextDiff()
-      ),
-      atom.commands.add(editorElement, 'git-diff:move-to-previous-diff', () =>
-        this.moveToPreviousDiff()
-      ),
-      atom.config.onDidChange('git-diff.showIconsInEditorGutter', () =>
-        this.updateIconDecoration()
-      ),
-      atom.config.onDidChange('editor.showLineNumbers', () =>
-        this.updateIconDecoration()
-      ),
-      editorElement.onDidAttach(() => this.updateIconDecoration()),
-      this.editor.onDidDestroy(() => {
-        this.cancelUpdate();
-        this.removeDecorations();
-        this.subscriptions.dispose();
-      })
+      atom.project.onDidChangePaths(subscribeToRepository)
     );
+  }
+
+  /**
+   * @describe Handles tear down of destructables and subscriptions.
+   *   Does not handle release of memory. This method should only be called
+   *   just before this object is freed, and should only tear down the main
+   *   object components that are guarunteed to exist at all times.
+   */
+  destroy() {
+    // This entire object will be free soon, no need to release here.
+    this.subscriptions.dispose();
+    this.destroyChildren();
+  }
+
+  /**
+   * @describe Destroys this objects children (non-freeing), it's intended
+   *   to be an ease-of use function for maintaing this object. This method
+   *   should only tear down objects that are selectively allocated upon
+   *   repository discovery.
+   *
+   *   Example: this.diffs only exists when we have a repository.
+   */
+  destroyChildren() {
+    if (this._animationId) cancelAnimationFrame(this._animationId);
```

Comment:
Please use `{}`. This code is not debuggable and hard to see the sentence

Reply:
To be clear, you really didn't need to comment on every single one of these. I got the point after the first one.